---
title: Ausnahmebehandlung
description: Grundlagen der Ausnahmebehandlung in F# sowie Links zu Ausnahmebehandlung, Ausdrücke und Funktionen.
ms.date: 05/16/2016
ms.openlocfilehash: 187236ca380c7de0b3714160f6c3703f8fb93d5a
ms.sourcegitcommit: 9b552addadfb57fab0b9e7852ed4f1f1b8a42f8e
ms.translationtype: MT
ms.contentlocale: de-DE
ms.lasthandoff: 04/23/2019
ms.locfileid: "62025092"
---
# <a name="exception-handling"></a><span data-ttu-id="84b1a-103">Ausnahmebehandlung</span><span class="sxs-lookup"><span data-stu-id="84b1a-103">Exception Handling</span></span>

<span data-ttu-id="84b1a-104">Dieser Abschnitt enthält Informationen über die Unterstützung für die Ausnahmebehandlung in der Sprache F#.</span><span class="sxs-lookup"><span data-stu-id="84b1a-104">This section contains information about exception handling support in the F# language.</span></span>

## <a name="exception-handling-basics"></a><span data-ttu-id="84b1a-105">Grundlagen der Ausnahmebehandlung</span><span class="sxs-lookup"><span data-stu-id="84b1a-105">Exception Handling Basics</span></span>
<span data-ttu-id="84b1a-106">Die Ausnahmebehandlung ist die Standardmethode zum Behandeln von Fehlerbedingungen in .NET Framework.</span><span class="sxs-lookup"><span data-stu-id="84b1a-106">Exception handling is the standard way of handling error conditions in the .NET Framework.</span></span> <span data-ttu-id="84b1a-107">Daher muss jede .NET-Sprache diesen Mechanismus unterstützen, einschließlich F#.</span><span class="sxs-lookup"><span data-stu-id="84b1a-107">Thus, any .NET language must support this mechanism, including F#.</span></span> <span data-ttu-id="84b1a-108">Eine *Ausnahme* ist ein Objekt, das Information über einen Fehler kapselt.</span><span class="sxs-lookup"><span data-stu-id="84b1a-108">An *exception* is an object that encapsulates information about an error.</span></span> <span data-ttu-id="84b1a-109">Wenn Fehler auftreten, werden Ausnahmen ausgelöst, und die reguläre Ausführung wird beendet.</span><span class="sxs-lookup"><span data-stu-id="84b1a-109">When errors occur, exceptions are raised and regular execution stops.</span></span> <span data-ttu-id="84b1a-110">Stattdessen sucht die Laufzeit nach einem passenden Handler für die Ausnahme.</span><span class="sxs-lookup"><span data-stu-id="84b1a-110">Instead, the runtime searches for an appropriate handler for the exception.</span></span> <span data-ttu-id="84b1a-111">Die Suche beginnt in der aktuellen Funktion und wird im Stapel in den höheren Ebenen der Aufrufer fortgesetzt, bis ein entsprechender Handler gefunden wird.</span><span class="sxs-lookup"><span data-stu-id="84b1a-111">The search starts in the current function, and proceeds up the stack through the layers of callers until a matching handler is found.</span></span> <span data-ttu-id="84b1a-112">Anschließend wird der Handler ausgeführt.</span><span class="sxs-lookup"><span data-stu-id="84b1a-112">Then the handler is executed.</span></span>

<span data-ttu-id="84b1a-113">Zudem führt die Laufzeit beliebigen Code in `finally`-Blöcken aus, wenn der Stapel entladen wird, um sicherzustellen, dass Objekte ordnungsgemäß während des Endladungsprozesses bereinigt werden.</span><span class="sxs-lookup"><span data-stu-id="84b1a-113">In addition, as the stack is unwound, the runtime executes any code in `finally` blocks, to guarantee that objects are cleaned up correctly during the unwinding process.</span></span>

## <a name="related-topics"></a><span data-ttu-id="84b1a-114">Verwandte Themen</span><span class="sxs-lookup"><span data-stu-id="84b1a-114">Related Topics</span></span>

|<span data-ttu-id="84b1a-115">Titel</span><span class="sxs-lookup"><span data-stu-id="84b1a-115">Title</span></span>|<span data-ttu-id="84b1a-116">Beschreibung</span><span class="sxs-lookup"><span data-stu-id="84b1a-116">Description</span></span>|
|-----|-----------|
|[<span data-ttu-id="84b1a-117">Ausnahmetypen</span><span class="sxs-lookup"><span data-stu-id="84b1a-117">Exception Types</span></span>](exception-types.md)|<span data-ttu-id="84b1a-118">Beschreibt, wie ein Ausnahmetyp deklariert wird</span><span class="sxs-lookup"><span data-stu-id="84b1a-118">Describes how to declare an exception type.</span></span>|
|[<span data-ttu-id="84b1a-119">Ausnahmen: Die `try...with` Ausdruck</span><span class="sxs-lookup"><span data-stu-id="84b1a-119">Exceptions: The `try...with` Expression</span></span>](the-try-with-expression.md)|<span data-ttu-id="84b1a-120">Beschreibt das Sprachkonstrukt, das Ausnahmebehandlung unterstützt</span><span class="sxs-lookup"><span data-stu-id="84b1a-120">Describes the language construct that supports exception handling.</span></span>|
|[<span data-ttu-id="84b1a-121">Ausnahmen: Die `try...finally` Ausdruck</span><span class="sxs-lookup"><span data-stu-id="84b1a-121">Exceptions: The `try...finally` Expression</span></span>](the-try-finally-expression.md)|<span data-ttu-id="84b1a-122">Beschreibt das Sprachkonstrukt, mit dem Sie Bereinigungscode auszuführen können, während der Stapel entladen wird, wenn eine Ausnahme ausgelöst wird</span><span class="sxs-lookup"><span data-stu-id="84b1a-122">Describes the language construct that enables you to execute clean-up code as the stack unwinds when an exception is thrown.</span></span>|
|[<span data-ttu-id="84b1a-123">Ausnahmen: Die `raise`-Funktion</span><span class="sxs-lookup"><span data-stu-id="84b1a-123">Exceptions: the `raise` Function</span></span>](the-raise-Function.md)|<span data-ttu-id="84b1a-124">Beschreibt, wie ein Ausnahmeobjekt ausgelöst wird</span><span class="sxs-lookup"><span data-stu-id="84b1a-124">Describes how to throw an exception object.</span></span>|
|[<span data-ttu-id="84b1a-125">Ausnahmen: Die `failwith` Funktion</span><span class="sxs-lookup"><span data-stu-id="84b1a-125">Exceptions: The `failwith` Function</span></span>](the-failwith-function.md)|<span data-ttu-id="84b1a-126">Beschreibt, wie eine allgemeine F#-Ausnahme generiert wird</span><span class="sxs-lookup"><span data-stu-id="84b1a-126">Describes how to generate a general F# exception.</span></span>|
|[<span data-ttu-id="84b1a-127">Ausnahmen: Die `invalidArg` Funktion</span><span class="sxs-lookup"><span data-stu-id="84b1a-127">Exceptions: The `invalidArg` Function</span></span>](the-invalidArg-function.md)|<span data-ttu-id="84b1a-128">Beschreibt, wie eine ungültige Argumentausnahme generiert wird</span><span class="sxs-lookup"><span data-stu-id="84b1a-128">Describes how to generate an invalid argument exception.</span></span>|